src/test/java/uk/ac/ebi/eva/pipeline/io/readers/VcfHeaderReaderTest.java

```diff
@@ -48,6 +51,19 @@
     @Rule
     public PipelineTemporaryFolderRule temporaryFolderRule = new PipelineTemporaryFolderRule();
 
+    @Test(expected = DuplicateSamplesFoundException.class)
+    public void testDuplicateSamples() throws Exception {
+        File input = getResource(INPUT_FILE_PATH_DUPLICATES);
+
+        VariantStudy.StudyType studyType = VariantStudy.StudyType.COLLECTION;
+        VariantSource.Aggregation aggregation = VariantSource.Aggregation.NONE;
+
+        VcfHeaderReader headerReader = new VcfHeaderReader(input, FILE_ID, STUDY_ID, STUDY_NAME,
+                studyType, aggregation);
```

Comment:
I noticed the method below does the same as this one, but others below use the constants. I think the code would look more clear if we use directly the constants; could you please also change the method right below?

Reply:
Could you elaborate? I am not sure if I understand what you mean here. I thought I am using constants the same way as other methods.

Reply:
Upon further revision I think what you requested is that I remove the indirection through the "studyType" and "aggregation" locally scoped variables for VariantStudy.StudyType.COLLECTION; and  VariantSource.Aggregation.NONE; respectively.

I initially took the indirect route as the fully qualified constant names are rather verbose and appear multiple times ( in the setup and final assertions ). Additionally these benefit from enhanced visibility of being set once in the test preamble. 

One compromise would be to adopt the same convention as that used for FILE_ID , STUDY_ID, etc. 

That is :  
Declare a set of private constants in the Test class and used these as indirections to the fully qualified constants. This has several benefits:
            - static final fields won't take up extra stack space per test 
            - the compiler will likely eliminate the intermediate storage anyway 
            - the developers can quickly asses and change the studyType and aggregation if required

Could you please confirm if you would prefer the private static field approach , or just the one that is established in the final test?

Reply:
Oops, you were right in using the indirections. I didn't notice the asserts below were using them.

Reply:
That's alright, thanks! I changed it back.